### PR TITLE
Added LC Daily Question Schedule

### PR DESCRIPTION
**Pull request by ashtongoh**
It creates a cron job to post the LC daily question at 8.01AM every day.

/startDailyLCSchedule
/stopDailyLCSchedule

Will definitely need to change it to an admin-only function, I guess?

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -247,4 +248,86 @@ function getCountStr(count) {
...
+// Definitely need to change this to an admin-only command
+bot.onText(/\/startDailyLCSchedule/, (msg) => {
+  const chatId = msg.chat.id;
+  const reply = `Starting daily LC schedule.`;
+  bot.sendMessage(chatId, reply);
+  console.log('start cron job');
+  // Just for testing every 5 seconds
+  // cronJob = cron.schedule('*/5 * * * * *', () => {
+  // Posts a daily question at 8:01AM 
+  cronJob = cron.schedule('01 8 * * *', () => {
```
Comment (paradite) [inline index.js:301]: add a command to check the status of `cronJob`, whether it is already scheduled or stopped or does not exist yet.

File: package.json [hunk 2]
```diff
@@ -11,6 +11,7 @@
   "dependencies": {
     "axios": "^1.1.3",
     "dayjs": "^1.11.6",
+    "node-cron": "^3.0.2",
```
Comment (paradite) [inline package.json:14]: remove `package-lock.json` file because we use yarn here.

File: index.js [hunk 3]
```diff
@@ -247,4 +248,86 @@ function getCountStr(count) {
...
+                freqBar
+                frontendQuestionId: questionFrontendId
+                isFavor
+                paidOnly: isPaidOnly
+                status
+                title
+                titleSlug
+                hasVideoSolution
+                hasSolution
+                topicTags {
```
Comment (paradite) [inline index.js]: remove unused fields from the query

File: index.js [hunk 4]
```diff
@@ -247,4 +248,86 @@ function getCountStr(count) {
...
+  cronJob = cron.schedule('01 8 * * *', () => {
+    getLCQuestion()
+      .then((response) => {
+        const data = response.data.data.activeDailyCodingChallengeQuestion;
+        const date = data.date;
+        const question = data.question;
+        const title = question.title;
+        const link = 'https://leetcode.com' + data.link;
+        const difficulty = question.difficulty;
+        const tags = question.topicTags.map((tag) => tag.name).join(', ');
```
Comment (paradite) [inline index.js]: remove tags since it might contain hints on how to solve the question

File: index.js (hunk 1 above)
Comment (hahaashton1) [reply index.js:301]: Created new variable to hold the status, cronStatus

**Comment by ashtongoh** [comment]
Changed code as per comments.

Also added new function called /checkDailyLCSchedule to check for status of cronJob.

**Pull request merged**